test/datetime.test.js

```diff
@@ -17,6 +17,7 @@ describe('MySQL datetime handling', function() {
     married: Boolean,
     dob: {type: 'DateString'},
     createdAt: {type: Date, default: Date},
+    lastLogon: {type: Date, precision: 3, default: Date},
```

Comment:
Are you sure? I ran the tests again and `dateOptions()` is hit twice for my specific test. The first time for property createdAt (skipping the if at [migration.js:914](https://github.com/joostdebruijn/loopback-connector-mysql/blob/c3a336c18dd9cebe1483d1fbe5b670a0496e26cf/lib/migration.js#L914)), resulting in a DATETIME. The second time for the property lastLogon (hitting the if at [migration.js:914](https://github.com/joostdebruijn/loopback-connector-mysql/blob/c3a336c18dd9cebe1483d1fbe5b670a0496e26cf/lib/migration.js#L914)) and resulting in a DATETIME(3).

Please note that the Loopback-type 'Date' maps on 'DATETIME' by default, see [documentation](https://loopback.io/doc/en/lb3/MySQL-connector.html#type-mappings) and [migration.js:725](https://github.com/joostdebruijn/loopback-connector-mysql/blob/c3a336c18dd9cebe1483d1fbe5b670a0496e26cf/lib/migration.js#L725). The method `dateOptions()` handles the columnType and not the type as in Loopback-type.